Objects/typeobject.c

```diff
@@ -5616,6 +5616,16 @@ _Py_type_getattro_impl(PyTypeObject *type, PyObject *name, int * suppress_missin
         res = meta_get(meta_attribute, (PyObject *)type,
                        (PyObject *)metatype);
         Py_DECREF(meta_attribute);
+        if (res && PyUnicode_EqualToUTF8(name, "__class_getitem__")) {
```

Comment:
This makes it so the deprecation gets triggered if you use regular attribute access to access the attribute, but I don't think that's right:

```
>>> class M(type):
...     def __class_getitem__(*args):
...         return "CGI"
...         
>>> class A(metaclass=M): pass
... 
>>> A.__class_getitem__
<python-input-7>:1: DeprecationWarning: Accessing __class_getitem__ on a metaclass is deprecated since 3.14 and will be removed in 3.16. Instead, define a regular __getitem__ method on a metaclass, if you need this behavior.
  A.__class_getitem__
<bound method M.__class_getitem__ of <class '__main__.M'>>
```

Instead, the deprecation should trigger only if the lookup is triggered by the subscript operator, like in this example:

```
>>> A["x"]
<python-input-8>:1: DeprecationWarning: Accessing __class_getitem__ on a metaclass is deprecated since 3.14 and will be removed in 3.16. Instead, define a regular __getitem__ method on a metaclass, if you need this behavior.
  A["x"]
'CGI'
```

Reply:
I don't think that there's any clean way to do that. We don't have any info about that in `_Py_type_getattro_impl` (nor in its public version). 

Passing this info would require either a global state or a new parameter in several functions (some of which are public).

Maybe I am missing something?

I don't think that `.__class_getitem__` access should really bother us, because:
1. Users should not access magic method directly
2. Warning is valid in *some* cases